io/src/lib.rs

```diff
@@ -65,6 +79,27 @@ pub struct Take<'a, R: Read + ?Sized> {
     remaining: u64,
 }
 
+impl<'a, R: Read + ?Sized> Take<'a, R> {
+    #[cfg(any(feature = "alloc", feature = "std"))]
+    #[inline]
+    fn read_to_end(&mut self, buf: &mut Vec<u8>) -> Result<usize> {
```

Comment:
I thought this would be `pub`. Either way is fine.

Reply:
Do you have an opinion @apoelstra? ?ooks like I thought it would be too otherwise I wouldn't have put `inline` on it.

Reply:
No real opinion; I also thought it'd be `pub` but to me the point of it being `pub` was to make it show up in search (where its doccomment would warn that it uses a hardcoded limit and that you probably want to manually call `take`).

Given that we have a better solution using `alias` I see no need.